pynest/nest/tests/test_iaf_psc_alpha_ps.py

```diff
@@ -0,0 +1,190 @@
+  - models are identical with respect to currents and spikes when
+  ``tau_syn_ex == tau_syn_in``,
+  - models differ with respect to spikes if ``tau_syn_ex != tau_syn_in``.
+
+Details:
+  The models are compared and we assess that the difference between the
+  recorded variables and the reference is smaller than a given tolerance.
+"""
+
+
+class IAFPreciseTestCase(unittest.TestCase):
+    """
+    Check the coherence between new and previous implementations.
+    """
+
+    def setUp(self):
+        '''
+        Clean up and initialize NEST before each test.
+        '''
+        msd = 123456
+        self.resol = 0.01
+        nest.ResetKernel()
+        N_vp = nest.GetKernelStatus(['total_num_virtual_procs'])[0]
+        pyrngs = [np.random.RandomState(s) for s in range(msd, msd + N_vp)]
+        nest.SetKernelStatus({
+            'resolution': self.resol, 'grng_seed': msd + N_vp,
+            'rng_seeds': range(msd + N_vp + 1, msd + 2 * N_vp + 1)})
+
+    def compute_difference(self, multimeters, params, reference, recordables):
+        '''
+        Compute the relative differences between the values recorded by the
+        multimeter and those of the reference (recorded at same times).
+
+        Parameters
+        ----------
+        multimeters : dict of tuples
+            Dictionary containing the model name as key and the GID of the
+            associated multimeter as value.
+        params : dict
+            Parameters used for the models.
+        reference : dict
+            Reference arrays (one per entry in `recordables`).
+        recordables : list of strings
+            List of recordables that will be compared.
+
+        Returns
+        -------
+        rel_diff : dict of dict of doubles
+            Relative differences between recorded data and reference (one dict
+            per model, containing one value per entry in `recordables`).
+        '''
+        rel_diff = {model: {} for model in multimeters.keys()}
+        V_lim = (params["V_th"] + params["V_peak"]) / 2.
+
+        for model, mm in iter(multimeters.items()):
+            dmm = nest.GetStatus(mm, "events")[0]
+            for record in recordables:
+                # ignore places where a divide by zero would occur
+                rds = np.abs(reference[record] - dmm[record])
+                nonzero = np.where(~np.isclose(reference[record], 0.))[0]
+                if np.any(nonzero):
+                    rds = rds[nonzero] / np.abs(reference[record][nonzero])
+                # ignore events around spike times for V if it diverges
+                if record == "V_m" and params["Delta_T"] > 0.:
+                    spiking = (dmm[record] > V_lim)
+                    rds = rds[~spiking]
+                rel_diff[model][record] = np.average(rds)
+        return rel_diff
+
+    def test_closeness_same_timescales(self):
+        '''
+        Compare models for tau_syn_ex == tau_syn_in (should be equal).
+        '''
+        Ie = 370.
```

Comment:
Insert empty line above.

Reply:
You want an empty line between the docstring and ``Ie``?! Damn... I hate that... :'( is it a NEST convention?

Reply:
I don't think we have an explicit convention, I just find it more legible. PEP257 requires an empty line behind class docstrings, but this is not the case here. Btw, PEP257 also wants double quotes instead of single quotes around docstrings ;)